api_tests/nodes/views/test_node_alternative_citations.py

```diff
@@ -0,0 +1,730 @@
+from nose.tools import *  # flake8: noqa
+
+from website.util import permissions
+
+from api.base.settings.defaults import API_BASE
+
+from tests.base import ApiTestCase
+
+from tests.factories import (
+    ProjectFactory,
+    RegistrationFactory,
+    AuthUserFactory,
+    AlternativeCitationFactory
+)
+
+class TestCreateAlternativeCitations(ApiTestCase):
+    def setUp(self):
+        super(TestCreateAlternativeCitations, self).setUp()
+        self.user = AuthUserFactory()
+        self.user_two = AuthUserFactory()
+        self.user_three = AuthUserFactory()
+
+        self.public = ProjectFactory(creator=self.user, is_public=True)
+        self.private = ProjectFactory(creator=self.user, is_public=False)
+        self.citation = AlternativeCitationFactory(name="name", text="text")
+        self.public.alternativeCitations.append(self.citation)
+        self.private.alternativeCitations.append(self.citation)
+        self.public.add_contributor(self.user_two, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+        self.private.add_contributor(self.user_two, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+        self.public.save()
+        self.private.save()
+        self.public_url = '/{}nodes/{}/citations/'.format(API_BASE, self.public._id)
+        self.private_url = '/{}nodes/{}/citations/'.format(API_BASE, self.private._id)
+        self.citation.public_url = self.public_url + '{}/'.format(self.citation._id)
+        self.citation.private_url = self.private_url + '{}/'.format(self.citation._id)
+
+    def payload(self, name=None, text=None, id=None):
+        payload = {'data': {
+            'type': 'citations',
+            'attributes': {
+                'name': name,
+                'text': text
+                }
+            }
+        }
+        if id is not None:
+            payload['data']['id'] = id
+        return payload
+
+    def test_add_citation_admin(self):
```

Comment:
Also, what is supposed to happen if someone tries to create a citation without a name or text? It would be good to have a test case for that situation as well.